clients/include/testing_local_matrix.hpp

```diff
@@ -715,4 +715,2109 @@ bool testing_local_matrix_set_data_ptr(Arguments argus)
+        EXPECT_EQ(mat.GetN(), n);
+    }
+
+    // Test AllocateELL
+    {
+        LocalMatrix<T> mat;
+        int            ell_nnz = 6, m = 3, n = 3, ell_max_row = 2;
+        EXPECT_NO_THROW(mat.AllocateELL("ELL", ell_nnz, m, n, ell_max_row));
+        EXPECT_EQ(mat.GetNnz(), ell_nnz);
+        EXPECT_EQ(mat.GetM(), m);
+        EXPECT_EQ(mat.GetN(), n);
+    }
+
+    // Test AllocateHYB
+    {
+        LocalMatrix<T> mat;
+        int            m = 3, n = 3;
+        int            ell_max_row = 2;
+        int            ell_nnz     = ell_max_row * m; // 2 * 3 = 6
+        int            coo_nnz     = 2;
+
+        EXPECT_NO_THROW(mat.AllocateHYB("HYB", ell_nnz, coo_nnz, ell_max_row, m, n));
+        EXPECT_EQ(mat.GetNnz(), ell_nnz + coo_nnz); // 8
+        EXPECT_EQ(mat.GetM(), m);
+        EXPECT_EQ(mat.GetN(), n);
+    }
+
+    // Test AllocateDENSE
+    {
+        LocalMatrix<T> mat;
+        int            m = 3, n = 3;
+        EXPECT_NO_THROW(mat.AllocateDENSE("DENSE", m, n));
+        EXPECT_EQ(mat.GetNnz(), m * n);
+        EXPECT_EQ(mat.GetM(), m);
+        EXPECT_EQ(mat.GetN(), n);
+    }
+}
+
+template <typename T>
+void testing_check_with_empty_matrix()
+{
+    LocalMatrix<T> empty_matrix;
+    // Check should pass without any issues
+    EXPECT_NO_THROW(empty_matrix.Check());
+    // Info should not throw an error
+    EXPECT_NO_THROW(empty_matrix.Info());
+}
+
+template <typename T>
+void testing_local_copy_from_async()
+{
+    auto           matrix = getTestMatrix<T>();
+    LocalMatrix<T> copy_matrix;
+    // CopyFromAsync should copy the matrix asynchronously (if supported)
+    EXPECT_NO_THROW(copy_matrix.CopyFromAsync(matrix));
+    EXPECT_NO_THROW(matrix.Sync());
+    EXPECT_EQ(copy_matrix.GetM(), matrix.GetM());
+    EXPECT_EQ(copy_matrix.GetN(), matrix.GetN());
+
+    int64_t nnz = matrix.GetNnz();
+
+    T* check_values_1 = new T[nnz];
+    T* check_values_2 = new T[nnz];
+
+    getMatrixVal(matrix, check_values_1);
+    getMatrixVal(copy_matrix, check_values_2);
+    for(int64_t i = 0; i < nnz; ++i)
+    {
+        EXPECT_EQ(check_values_1[i], check_values_2[i]);
+    }
+
+    delete[] check_values_1;
+    delete[] check_values_2;
+}
+
+template <typename T>
+void testing_local_update_values_csr()
+{
+    auto matrix = getTestMatrix<T>();
+
+    int64_t nnz = matrix.GetNnz();
+
+    T* new_values = new T[nnz];
+    for(int64_t i = 0; i < nnz; ++i)
+    {
+        new_values[i] = static_cast<T>(i + 10); // Fill with some values
+    }
+
+    // UpdateValuesCSR should update the values in the matrix
+    EXPECT_NO_THROW(matrix.UpdateValuesCSR(new_values));
+
+    T* check_values = new T[nnz];
+    getMatrixVal(matrix, check_values);
+    for(int64_t i = 0; i < nnz; ++i)
+    {
+        EXPECT_EQ(check_values[i], new_values[i]);
+    }
+
+    delete[] new_values;
+    delete[] check_values;
+}
+
+template <typename T>
+void testing_local_move_to_accelerator()
+{
+    auto matrix = getTestMatrix<T>();
```

Comment:
I think we should have a test that is a bit more complicated like:

    for(int i = 0; i < 10; i++)
    {
        EXPECT_NO_THROW(matrix.MoveToAccelerator());
    }
    for(int i = 0; i < 10; i++)
    {
        EXPECT_NO_THROW(matrix.MoveToHost());
    }
    for(int i = 0; i < 10; i++)
    {
        EXPECT_NO_THROW(matrix.MoveToHost());
        EXPECT_NO_THROW(matrix.MoveToAccelerator());
    }
    EXPECT_EQ(matrix.GetM(), 2);
    EXPECT_EQ(matrix.GetN(), 2);
This ensures that we can call the move repeatedly and alternating back and forth and it works correctly without crashing